src/controller/CHIPDeviceController.h

```diff
@@ -401,11 +401,13 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController,
      *   Note: Pairing process requires that the caller has registered PersistentStorageDelegate
      *         in the Init() call.
      *
-     * @param[in] remoteDeviceId        The remote device Id.
-     * @param[in] setUpCode             The setup code for connecting to the device
+     * @param[in] remoteDeviceId            The remote device Id.
+     * @param[in] setUpCode                 The setup code for connecting to the device
+     * @param[in] useOnlyOnNetworkDiscovery If true, perform on-network discovery only.
      */
-    CHIP_ERROR PairDevice(NodeId remoteDeviceId, const char * setUpCode);
-    CHIP_ERROR PairDevice(NodeId remoteDeviceId, const char * setUpCode, const CommissioningParameters & CommissioningParameters);
+    CHIP_ERROR PairDevice(NodeId remoteDeviceId, const char * setUpCode, bool useOnlyOnNetworkDiscovery = false);
```

Comment:
could we use some enum instead of a bool - generally I would prefer to not use bools as argument unless really obvious (like `setDebugEnabled(bool)` is obvious).

This could read `discoveryType = kDiscoveryNetworkOnly` or `discoveryType = DiscoveryType::kAll`.